src/Plugin/TestsiteBuilder/ConfigTemplateType/ViewFilter.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Drupal\testsite_builder\Plugin\TestsiteBuilder\ConfigTemplateType;
+
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\testsite_builder\ConfigTemplateMerge;
+use Drupal\views\ViewsData;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * View filter config template type plugin.
+ *
+ * @TestsiteBuilderConfigTemplateType(
+ *   id = "view_filter",
+ *   label = @Translation("View filter"),
+ *   description = @Translation("View filter config template type plugin.")
+ * )
+ */
+class ViewFilter extends Generic {
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The views data service.
+   *
+   * @var \Drupal\views\ViewsData
+   */
+  protected $viewsData;
+
+  /**
+   * Then entity field manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+   */
+  protected $entityFieldManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, ViewsData $views_data, EntityFieldManagerInterface $entity_field_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->entityTypeManager = $entity_type_manager;
+    $this->viewsData = $views_data;
+    $this->entityFieldManager = $entity_field_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static($configuration, $plugin_id, $plugin_definition, $container->get('entity_type.manager'), $container->get('views.views_data'), $container->get('entity_field.manager'));
+  }
+
+  /**
+   * Applies plugin specific configuration.
+   *
+   * @param string $entity_type
+   *   The entity type.
+   * @param string $bundle
+   *   The bundle.
+   * @param string $field_name
+   *   The field name.
+   * @param mixed $config
+   *   The plugin configuration.
+   *
+   * @return mixed
+   *   Returns plugin configuration or FALSE.
+   */
+  protected function applyPlugConfiguration(string $entity_type, string $bundle, string $field_name, $config) {
```

Comment:
This smells like it should also be made into a plugin, as soon as there are more plugin_ids we might want to handle, this gets messy.